calicoctl/Dockerfile.calicoctl.build-s390x

```diff
@@ -0,0 +1,8 @@
+FROM s390x/golang:1.8.3
```

Comment:
Hm, looking at the Makefile now I can't see that the `Dockerfile.calicoctl.build` series is ever used - looks like it's all using `calico/go-build` now.

So, I think we can remove this file.